pmd-dist/src/main/resources/scripts/pmd

```diff
@@ -209,4 +217,9 @@ cygwin_paths
 
 java_heapsize_settings
 
-java ${HEAPSIZE} ${PMD_JAVA_OPTS} $(jre_specific_vm_options) -cp "${classpath}" net.sourceforge.pmd.cli.PmdCli "$@"
+# Note: we want word-splitting happening on PMD_JAVA_OPTS and jre_specific_vm_options
+exec java \
+  ${HEAPSIZE:+"$HEAPSIZE"} \
```

Comment:
This avoids creating a empty argument (`""`) if HEAPSIZE is not defined. Only when HEAPSIZE is defined, `"$HEAPSIZE"` is used.

That was reason for the error from CI build "Error: Could not find or load main class "